### PR TITLE
feat(inbox): Add mark reviewed to issue activity (WOR-570)

### PR DESCRIPTION
**Pull request by scttcper**
When a user marks an issue as reviewed, add an activity item that can be found in issue details and org activity.

image

### CODE REVIEW

File: tests/sentry/models/test_groupinbox.py [hunk 1]
```diff
@@ -23,10 +30,13 @@ def test_remove_from_inbox(self, inbox_out):
...
             group=self.group, reason=GroupInboxReason.NEW.value
         ).exists()
-        remove_group_from_inbox(self.group)
+        remove_group_from_inbox(
+            self.group, user=self.user, action=GroupInboxRemoveAction.MARK_REVIEWED
+        )
         assert not GroupInbox.objects.filter(
             group=self.group, reason=GroupInboxReason.NEW.value
         ).exists()
+        assert Activity.objects.count() == 1
```
Comment (iProgramStuff) [inline tests/sentry/models/test_groupinbox.py:39]: I would go so far as to assert the activity type here

**Review by iProgramStuff — APPROVED** [review]
This basically looks good to me. The migration looks like it will be okay. I'm sure someone from @getsentry/owners-migrations will chime in soon.

File: src/sentry/models/groupinbox.py [hunk 2]
```diff
@@ -89,12 +96,20 @@ def remove_group_from_inbox(group, action=None, user=None):
         group_inbox = GroupInbox.objects.get(group=group)
         group_inbox.delete()
 
+        if action is GroupInboxRemoveAction.MARK_REVIEWED and user is not None:
+            Activity.objects.create(
+                project_id=group_inbox.group.project.id,
```
Comment (mattrobenolt) [inline src/sentry/models/groupinbox.py:101]:
```suggestion
                project_id=group_inbox.group.project_id,
```

File: src/sentry/models/groupinbox.py [hunk 3]
```diff
@@ -89,12 +96,20 @@ def remove_group_from_inbox(group, action=None, user=None):
...
+                group_id=group_inbox.group.id,
```
Comment (mattrobenolt) [inline src/sentry/models/groupinbox.py:102]:
```suggestion
                group_id=group_inbox.group_id,
```

File: src/sentry/migrations/0155_add_mark_reviewed_activity.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    # You'll also usually want to set this to `False` if you're writing a data
+    # migration, since we don't want the entire migration to run in one long-running
+    # transaction.
+    atomic = True
+
+    dependencies = [
+        ("sentry", "0154_groupedmessage_inbox_sort"),
+    ]
+
+    operations = [
```
Comment (mattrobenolt) [inline src/sentry/migrations/0155_add_mark_reviewed_activity.py:33]: @wedamija this translates to doing nothing against the database right? It's just an internal state change.

**Review by mattrobenolt — APPROVED** [review]

**Pull request merged**